src/main/java/com/moodybluez/enterprise/dao/DateDAO.java

```diff
@@ -0,0 +1,51 @@
+package com.moodybluez.enterprise.dao;
+
+import com.moodybluez.enterprise.dto.Date;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Component
+public class DateDAO implements IDateDAO{
```

Comment:
Created new DAO class. Added logic from service class which should have been actually in DAO (this) layer.
Citation - "The Data Access Object (DAO) pattern is a structural pattern that allows us to isolate the application/business layer from the persistence layer" - [Baeldung](https://www.baeldung.com/java-dao-pattern)